conda/repodata.py

```diff
@@ -0,0 +1,99 @@
+# (c) Continuum Analytics, Inc. / http://continuum.io
+# All Rights Reserved
+#
+# conda is distributed under the terms of the BSD 3-clause license.
+# Consult LICENSE.txt or http://opensource.org/licenses/BSD-3-Clause.
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+from .core import repodata as _impl
+from .connection import CondaSession
+from os.path import join
+
+
+class RepoData(object):
+    """This object represents all the package metainfo of a single channel."""
+
+    _data = {}
+
+    @staticmethod
+    def enable(url, name, priority, cache_dir=None):
+        RepoData._data[url] = RepoData(url, name, priority, cache_dir)
+
+    @staticmethod
+    def get(url):
+        return RepoData._data[url]
+
+    @staticmethod
+    def clean():
+        RepoData._data = {}
+
+    @staticmethod
+    def sync_all(use_cache=False):
+        try:
+            import concurrent.futures
+            with concurrent.futures.ThreadPoolExecutor(10) as e:
+                for rd in RepoData._data.values():
+                    e.submit(rd.sync(use_cache=use_cache, session=CondaSession()))
+        except (ImportError) as e:
+            for rd in RepoData._data.values():
+                rd.sync(use_cache=use_cache, session=CondaSession())
+
+    def __init__(self, url, name, priority, cache_dir=None):
```

Comment:
I admit I merely cloned from the existing code, where the above 3-tuple is used. Yeah, the main ID of the RepoData is the (canonical_channel_name, subdir) pair which forms the url.
But our current code also requires the other two (though ideally the RepoData itself shouldn't know or care about things like priorities.
I suppose the priority can be encoded using an OrderedDict for the RepoData._data, rather than a simple dict. Where is the name used ?